pkg/controller/operconfig/migration_test.go

```diff
@@ -0,0 +1,494 @@
+					},
+				},
+			},
+			expectedEgressIpList: nnsEgressIpsStrArrMult,
+		},
+		{
+			name: "Hostsubnet has manual config and netnamespace has one egressIP",
+			objects: []crclient.Object{
+				&v1.HostSubnet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testMigrationHost,
+					},
+					EgressIPs: []v1.HostSubnetEgressIP{"10.0.128.5"},
+					Host:      testMigrationHost,
+				},
+				&uns.Unstructured{
+					Object: map[string]interface{}{
+						"apiVersion": "network.openshift.io/v1",
+						"kind":       "NetNamespace",
+						"egressIPs":  nnsEgressIpsIfcArrSingle,
+						"netname":    testMigrationNamespace,
+						"metadata": map[string]interface{}{
+							"name": testMigrationNamespace,
+						},
+					},
+				},
+				&corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testMigrationHost,
+					},
+				},
+				&corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testMigrationNamespace,
+					},
+				},
+			},
+			expectedEgressIpList: nnsEgressIpsStrArrSingle,
+		},
+		{
+			name: "Hostsubnet has manual config and netnamespace has multiple egressIPs",
+			objects: []crclient.Object{
+				&v1.HostSubnet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testMigrationHost,
+					},
+					EgressIPs: []v1.HostSubnetEgressIP{"10.0.128.5", "10.0.128.6", "10.0.128.7"},
+					Host:      testMigrationHost,
+				},
+				&uns.Unstructured{
+					Object: map[string]interface{}{
+						"apiVersion": "network.openshift.io/v1",
+						"kind":       "NetNamespace",
+						"egressIPs":  nnsEgressIpsIfcArrMult,
+						"netname":    testMigrationNamespace,
+						"metadata": map[string]interface{}{
+							"name": testMigrationNamespace,
+						},
+					},
+				},
+				&corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testMigrationHost,
+					},
+				},
+				&corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testMigrationNamespace,
+					},
+				},
+			},
+			expectedEgressIpList: nnsEgressIpsStrArrMult,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			client := fake.NewFakeClient(tc.objects...)
+
+			egressIpList, err := convertSdnEgressIpToOvnEgressIp(context.Background(), client)
+			if err != nil {
+				t.Fatalf("convertSdnEgressIpToOvnEgressIp: %v", err)
+			}
+			// Collect all egressIP strings in a single slice
+			var egressIpValueList []string
+			for _, egressIp := range egressIpList {
+				egressIpsSlice := egressIp.Object["spec"].(map[string]interface{})["egressIPs"].([]interface{})
+				egressIpsStringSlice := make([]string, len(egressIpsSlice))
+				for i := range egressIpsSlice {
+					egressIpsStringSlice[i] = egressIpsSlice[i].(string)
+				}
+				egressIpValueList = append(egressIpValueList, egressIpsStringSlice...)
+			}
+
+			// Check if values match exactly
+			less := func(a, b string) bool { return a < b }
+			if !(cmp.Diff(egressIpValueList, tc.expectedEgressIpList, cmpopts.SortSlices(less)) == "") {
```

Comment:
gomega can do this, and is a dependency we already have.
Initialize a gomega instance with t at the beginning of the test case run:
```
import (
    . "github.com/onsi/gomega"
)

...

    t.Run(tc.name, func(t *testing.T) {
        g := gomega.WithT(t)

    ...
```

and then you can use

```
g.Expect(egressIpValueList).To(ConsistOf(tc.expectedEgressIpList...), "optional error explanation")
```